controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1855,6 +1859,27 @@ func (r *BareMetalHostReconciler) createHostFirmwareSettings(info *reconcileInfo
 	return nil
 }
 
+func (r *BareMetalHostReconciler) hostUpdatePolicySetOwnerReference(info *reconcileInfo) error {
+	// NOTE(janders) the goal here is to ensure that the controller reads the hup resource and adds OwnerReference to it
+	hup := &metal3api.HostUpdatePolicy{}
+	if err := r.Get(info.ctx, info.request.NamespacedName, hup); err != nil {
+		// Error reading the object
+		return fmt.Errorf("could not load hostUpdatePolicy resource due to %w", err)
+	}
+	if !ownerReferenceExists(info.host, hup) {
+		if err := controllerutil.SetOwnerReference(info.host, hup, r.Scheme()); err != nil {
+			return errors.Wrap(err, "could not set bmh as owner for hostUpdatePolicy")
```

Comment:
nit: use fmt.Errorf here and below (Wrap is deprecated)

Reply:
https://pkg.go.dev/errors#Join